ktor-shared/ktor-resources/common/src/io/ktor/resources/serialization/Decoders.kt

```diff
@@ -28,8 +28,10 @@ internal class ParametersDecoder(
         while (parameterNames.hasNext()) {
             currentName = parameterNames.next()
             val elementIndex = descriptor.getElementIndex(currentName)
-            val isPrimitive = descriptor.getElementDescriptor(elementIndex).kind is PrimitiveKind
-            if (!isPrimitive || parameters.contains(currentName)) {
+            val elementDescriptorKind = descriptor.getElementDescriptor(elementIndex).kind
```

Comment:
I would extract it to extension

Reply:
that's the only place where it's used, so not much point in it